core/src/main/java/com/datastax/oss/driver/internal/core/channel/ProtocolInitHandler.java

```diff
@@ -294,6 +294,8 @@ void onResponse(Message response) {
         }
       } catch (AuthenticationException e) {
         fail(e);
+      } catch (Throwable t) {
+        fail("Unexpected exception at step " + step, t);
```

Comment:
I changed this at the last minute because it could in theory be something else than an authentication exception. I've checked that it's properly surfaced to the thread that creates the session.